srcCpp/perftest_publisher.cxx

```diff
@@ -1850,12 +1889,28 @@ int perftest_cpp::Publisher()
                     // back the LENGTH_CHANGED_SIZE message
                     message.latency_ping = num_pings % _NumSubscribers;
 
-                    while (announcement_reader_listener->announced_subscribers > 0) {
+                    /*
+                     * If the Throughput topic is reliable, we can send the packet and do
+                     * a wait for acknowledgements. However, if the Throughput topic is
+                     * Best Effort, wait_for_acknowledgments() will return inmediately.
+                     * This would cause that the Send() would be exercised too many times,
+                     * in some cases causing the network to be flooded, a lot of packets being
+                     * lost, and potentially CPU starbation for other processes.
+                     * We can prevent this by adding a small sleep() if the test is best
+                     * effort.
+                     */
+                    announcement_reader_listener->subscriber_list.clear();
+                    while ((int)announcement_reader_listener->subscriber_list.size()
```

Comment:
We do:  message.size = LENGTH_CHANGED_SIZE;
We clear the list of subscribers. and then we send the messages, since these are answered via the announcement writer of the Perftest subscribers, that will trigger that we will reload the list, and once that list is equal to the number of expected subscribers, that means all the Subscribers know about the change in datalen